RecoTauTag/RecoTau/plugins/DPFIsolation.cc

```diff
@@ -68,21 +65,18 @@ class DPFIsolation : public deep_tau::DeepTauBase {
         descriptions.add("DPFTau2016v0", desc);
     }
 
-    explicit DPFIsolation(const edm::ParameterSet& cfg) :
-        DeepTauBase(cfg, GetOutputs()),
+    explicit DPFIsolation(const edm::ParameterSet& cfg,const deep_tau::DeepTauCache* cache) :
+        DeepTauBase(cfg, GetOutputs(), cache),
         pfcand_token(consumes<pat::PackedCandidateCollection>(cfg.getParameter<edm::InputTag>("pfcands"))),
-        vtx_token(consumes<reco::VertexCollection>(cfg.getParameter<edm::InputTag>("vertices")))
+        vtx_token(consumes<reco::VertexCollection>(cfg.getParameter<edm::InputTag>("vertices"))),
+        graphVersion(cfg.getParameter<unsigned>("version"))
     {
-        if(graphName.find("v0.pb") != std::string::npos)
-            graphVersion = 0;
-        else if(graphName.find("v1.pb") != std::string::npos)
-            graphVersion = 1;
-        else
+        if(!(graphVersion == 1 || graphVersion == 0 ))
```

Comment:
This part was removed because the new files have at the end some the "quantized" suffix, and to solve this was added a new parameter in the cofig called graphVersion for each of the graph files, so I think it shouldn't be mishmash

Reply:
I find binding of the version to the file name to be not flexible. For example, one can have version 1.X, which is a new training with the same inputs, so no need to change the c++ code. Another example files with quantization or memory mapping: it looks more ergonomic to have corresponding indications in the file names as suffices.
On the other hand, I agree that some protection checks can be added. For example that the graph has the number of inputs which are expected for the given version.

Reply:
Concerning previous system of "version recognition": it can be always updated to use a more sophisticated "version patter" in file name. For me it is still more safe than separate setting of file name and version as version defines inputs used which should match with what is in file. Then, I agree that an automatic way of finding version based e.g. on number of inputs if accessible from a file will be even better. So, is it possible to read it from file? And if so, how much it costs in terms of memory?

Reply:
Ok, fine with me to go back to the old "version recognition" + additional consistency check for the input size. There is no memory costs attached to it. However, I propose that we finish this PR with the currently implemented version check and once it is confirmed that quantized files can be used as the basilene, we'll do other PR removing 'Q' and alternative configs, changing file names, and rolling back to the file name based "version recognition", etc. Are you agree?

Reply:
OK, fine with me. Please introduce only style comments, i.e. about datamember names, removing code commented out, etc.